bgpd/bgp_mplsvpn.c

```diff
@@ -1564,6 +1568,17 @@ void vpn_handle_router_id_update(struct bgp *bgp, bool withdraw,
 	struct ecommunity *ecom;
 	vpn_policy_direction_t idir, edir;
 
+	/*
+	 * Router-id change that is not explicitly configured
+	 * (a change from zebra, frr restart for example)
+	 * should not replace a configured vpn RD/RT.
+	 */
+	if (!is_config) {
+		if (debug)
+			zlog_debug("skipping non explicit router-id change");
```

Comment:
Can you add `__func__` for every zlog_debug output?

Reply:
We will add it.

Reply:
Done